server/src/domain/auth/auth.service.ts

```diff
@@ -251,11 +251,26 @@ export class AuthService {
         storageLabel = null;
       }
 
+      let defaultQuota: number | null = config.oauth.defaultStorageQuota as keyof OAuthProfile as number;
+      if (typeof defaultQuota !== 'number') {
+        defaultQuota = null;
+      }
+
+      let storageQuota: number | null = profile[config.oauth.storageQuotaClaim as keyof OAuthProfile] as number;
+      if (typeof storageQuota !== 'number') {
+        storageQuota = defaultQuota;
+      }
+
+      if (typeof storageQuota == 'number' && storageQuota == 0) {
+        storageQuota = null;
+      }
+
       const userName = profile.name ?? `${profile.given_name || ''} ${profile.family_name || ''}`;
       user = await this.userCore.createUser({
         name: userName,
         email: profile.email,
         oauthId: profile.sub,
+        quotaSizeInBytes: storageQuota ? storageQuota * 1_073_741_824 : null,
```

Comment:
I wasn't sure if/how I should use https://github.com/samholton/immich/blob/feature/add-quota-to-oauth-claim/web/src/lib/utils/byte-converter.ts

Reply:
You shouldn't import code from web in the server. We have these constants defined on the server side you can use:

https://github.com/immich-app/immich/blob/55ecfafa82e692f56bab40b7c559eface327158f/server/src/domain/domain.util.ts#L116

Reply:
We should discuss how we want to accept input for the claim. Id be fine with just importing the bytes package and delegating all conversations to that.

https://www.npmjs.com/package/bytes

Reply:
I'm open to suggestions. Originally I was thinking a string such as "50 GiB" but decided to keep it consistent with what you would enter on the edit user screen and used a number (in GiB). Are you suggesting I use a string and just pass it to `bytes`?

Reply:
That is one option. I'm open to suggestions. Making it work the same way as the web seems like a good idea. So it would need to be a number and we'd assume it is in GiB?

Reply:
Yeah, I think that makes sense. And if web is ever changed to be open ended as a string, this could be as well. I will update to bring in bytes, keep the type as number, and pass it to bytes with assumed GiB for conversion.

Reply:
If we're assuming it is GiB we don't need to import the bytes package